zebra-consensus/src/checkpoint.rs

```diff
@@ -135,24 +176,251 @@ where
 pub fn init<S>(
```

Comment:
I think it would be better if this were just `CheckpointVerifier::new`, both because that's a more common idiom for constructors and because it means that we don't have to erase the return type.  This means we can include it in other services without having to add an extra generic parameter, and we don't have to write horrible nested `impl Trait` clauses.

Reply:
Done in fee7a62f704136da060b0123e72db4645228dfb7.